crates/store/src/state.rs

```diff
@@ -354,6 +381,15 @@ impl State {
         // Signals the write lock has been acquired, and the transaction can be committed
         let (inform_acquire_done, acquire_done) = oneshot::channel::<()>();
 
+        // Extract account updates with deltas before block is moved into async task
+        // We'll use these deltas to update the SmtForest without DB roundtrips
+        let account_updates: Vec<_> = block
+            .body()
+            .updated_accounts()
+            .iter()
+            .map(|update| (update.account_id(), update.details().clone()))
+            .collect();
```

Comment:
Since we are doing the iteration here, might as well filter out private account updates - e.g., this could look something like:
```rust
let public_account_deltas: Vec<_> = block
    .body()
    .updated_accounts()
    .iter()
    .filter_map(|update| match update.details() {
        AccountUpdateDetails::Private => None,
        AccountUpdateDetails::Delta(delta) => Some(delta.clone()),
    })
    .collect();
```

Reply:
Removed, was renamed in next